src/main/java/it/gov/pagopa/pu/pagopapayments/connector/pagopa/gpd/GpdServiceImpl.java

```diff
@@ -2,29 +2,111 @@
-    client.createPosition(apiKey, organizationfiscalcode,paymentPositionModel);
+    registryLogger.execute(
+      getRegistryContextDataFromPaymentPositionModel(
+        organizationfiscalcode,
+        RegistryEventType.createPosition,
+        paymentPositionModel.getIupd(),
+        paymentPositionModel
+      ),
+      paymentPositionModel,
+      () -> {
+        client.createPosition(apiKey, organizationfiscalcode,paymentPositionModel);
+        return Triple.of(null, null, RegistryOutcome.OK);
+      },
+      e -> {
+        log.error(e.getMessage(), e);
+        return null;
+      }
+    );
   }
 
   @Override
   public void paUpdatePosition(String apiKey, String organizationfiscalcode, String iupd, PaymentPositionModel paymentPositionModel) {
-    client.updatePosition(apiKey, organizationfiscalcode,iupd, paymentPositionModel);
+    registryLogger.execute(
+      getRegistryContextDataFromPaymentPositionModel(
+        organizationfiscalcode,
+        RegistryEventType.updatePosition,
+        iupd,
+        paymentPositionModel
+      ),
+      paymentPositionModel,
+      () -> {
+        client.updatePosition(apiKey, organizationfiscalcode, iupd, paymentPositionModel);
+        return Triple.of(null, null, RegistryOutcome.OK);
+      },
+      e -> {
+        log.error(e.getMessage(), e);
+        return null;
+      }
+    );
   }
 
   @Override
   public void paDeletePosition(String apiKey, String organizationfiscalcode, String iupd) {
-    client.deletePosition(apiKey, organizationfiscalcode,iupd);
+    RegistryContextData contextData = RegistryContextData.builder()
+      .orgFiscalCode(organizationfiscalcode)
+      .brokerStationId(null) // @TODO: no broker station found on payment position model
+      .pspId(null) // @TODO: no psp found on payment position model
+      .pspChannelId(null) // @TODO: no psp channel found on payment position model
+      .paymentMethod(null) // @TODO: being a position, does it have a payment method?
+      .ccp(null) // @TODO: i guess the receipt is not available at this point
+      .eventType(RegistryEventType.deletePosition)
+      .iuv(Utilities.nav2Iuv(iupd))
+      .build();
+
+    registryLogger.execute(
+      contextData,
+      iupd,
+      () -> {
+        client.deletePosition(apiKey, organizationfiscalcode, iupd);
+        return Triple.of(null, null, RegistryOutcome.OK);
+      },
+      e -> {
+        log.error(e.getMessage(), e);
+        return null;
+      }
+    );
+  }
+
+  private RegistryContextData getRegistryContextDataFromPaymentPositionModel(
+    String organizationFiscalCode,
+    RegistryEventType eventType,
+    String iupd,
+    PaymentPositionModel paymentPositionModel
+  ) {
+    return RegistryContextData.builder()
+      .orgFiscalCode(organizationFiscalCode)
+      .brokerStationId(null) // @TODO: no broker station found on payment position model
```

Comment:
@serdimic do you think we could avoid to store this? the station does make sense just when the Nodo is calling PU...